### PR TITLE
LPD-20985 - use user's languageId rather than the default locale's when adding a user

### PR DESCRIPTION
**Pull request by wanderlast**
Hi @smottal ,

https://liferay.atlassian.net/browse/LPD-20985, tied to LPP.

This is a quick fix to bring the POST method and a handful of other calls I noticed in line with the PATCH and GET methods. We want to use the user's languageId rather than the default preferred locale's per your investigation on feature request https://liferay.atlassian.net/browse/LPD-18602.

### CODE REVIEW

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java [hunk 1]
```diff
@@ -651,7 +652,7 @@ public UserAccount postAccountUserAccount(
 			_accountEntryUserRelService.addAccountEntryUserRel(
 				accountId, contextUser.getUserId(),
 				userAccount.getAlternateName(), userAccount.getEmailAddress(),
-				contextAcceptLanguage.getPreferredLocale(),
+				LocaleUtil.fromLanguageId(userAccount.getLanguageId()),
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java:655]: Can we use the old behavior as a fallback? If userAccount.getLanguageId is null use the contextAcceptLanguage.getPreferredLocale()

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by wanderlast** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java (hunk 1 above)
Comment (wanderlast) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java]: See: 3aa7c8e473cb3325b6828222eb60d3d50f7fec2b

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java [hunk 2]
```diff
@@ -1326,6 +1327,14 @@ private DTOConverterContext _getDTOConverterContext(long userId) {
 			contextUser);
 	}
 
+	private Locale _getLocale(String languageId) {
+		if (languageId != null) {
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java:1331]: please use Validator.isNull

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:2554484638

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**